File: my-folder/0045-jump-game-ii/solution.c
int jump(int* nums, int numsSize) {


    if (numsSize <= 1)
        return 0;

    int jumps = 0;
    int farthest = 0;
    int current_end = 0;

    for (size_t i = 0; i < numsSize; i++){
        if ((i + nums[i]) > farthest)
            farthest = (i + nums[i]);

        if (i == current_end) {
            jumps++;
            current_end = farthest;
            if (current_end >= (numsSize - 1))
                break;
        }
    }

    return jumps;
}

File: my-folder/0045-jump-game-ii/solution.rs
impl Solution {
    pub fn jump(nums: Vec<i32>) -> i32 {
        if nums.len() <= 1 {
            return 0;
        }

        let mut jumps = 0;
        let mut current_end = 0;
        let mut farthest = 0;

        for (i, &num) in nums.iter().enumerate() {
            farthest = farthest.max(i + num as usize);
            if i == current_end {
                jumps += 1;
                current_end = farthest;
                if current_end >= nums.len() - 1 {
                    break;
                }
            }
        }

        jumps
    }
}
